### PR TITLE
[SPARK] add debug facet to help resolving Spark integration issues

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
### Problem

Debugging openlineage-spark problems is tedious job. We would like to have debug facet that will collect automatically meaningful information when enabled. 

Closes: #2135 

### Solution

 * Create debug facet (more details in #2135 ), 
 * Facet is disabled by default, 
 * Allow enabling it thourgh SparkConf. 

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        buildSparkConfigDebugFacet(),
+        buildClasspathDebugFacet(),
+        buildSystemDebugFacet(),
+        buildLogicalPlanDebugFacet());
+  }
+
+  private SparkConfigDebugFacet buildSparkConfigDebugFacet() {
+    return SparkConfigDebugFacet.builder()
+        .extraListeners(getSparkConfOrNull("spark.extraListeners"))
+        .openLineageConfig(getOpenLineageConfig())
```
Comment (julienledem) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java:56]: shou.d we remove the API key or any secret?

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    return olContext
+        .getSparkSession()
+        .map(sparkSession -> sparkSession.catalog())
+        .map(catalog -> catalog.getClass().getCanonicalName())
+        .orElse(null);
+  }
+
+  private boolean isOnClassPath(String aClass) {
+    try {
+      this.getClass().getClassLoader().loadClass(aClass);
```
Comment (julienledem) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java:164]: isn't `Class.forName()` more appropriate here? Does it need to be in the same ClassLoader?

**Review by julienledem — APPROVED** [review]

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java (hunk 2 above)
Comment (pawel-big-lebowski) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java:164]: We're not consistent across the codebase and have both approaches. (https://stackoverflow.com/questions/8100376/class-forname-vs-classloader-loadclass-which-to-use-for-dynamic-loading gives more details on differences)

"getClassLoader" uses `system` classloader (which is overridable) while `Class.forName` initialises the class (runs static methods).

Comment (pawel-big-lebowski) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DebugRunFacetBuilderDelegate.java:164]: I've merged the issues but I would be happy to continue discussion on that which can possibly lead to other PR or issue created.

**Pull request merged**